aws-java-sdk-core/src/main/java/com/amazonaws/retry/PredefinedBackoffStrategies.java

```diff
@@ -102,8 +102,10 @@ public ExponentialBackoffStrategy(final int baseDelay,
         public long delayBeforeNextRetry(AmazonWebServiceRequest originalRequest,
                                          AmazonClientException exception,
                                          int retriesAttempted) {
+            long potentialWait = (1L << retriesAttempted) * baseDelay;
             return (retriesAttempted > MAX_RETRIES) ? maxBackoffTime :
-                    Math.min(((1 << retriesAttempted) * baseDelay), maxBackoffTime);
+                    (potentialWait < 0  ? maxBackoffTime :
```

Comment:
Reckon we can push this into a static private method that the various strategies could use?